server/pulp/server/controllers/repository.py

```diff
@@ -767,6 +767,11 @@ def sync(repo_id, sync_config_override=None, scheduled_call_id=None):
         raise
 
     else:
+        # Update the override config if it has changed
+        if check_override_config_change(repo_id, call_config):
```

Comment:
maybe also add the explanation of why the raw pymongo is used here "# Use raw pymongo not to fire the signal hander"